src/CoreWCF.Primitives/src/CoreWCF/Description/MessageContractHelper.cs

```diff
@@ -12,7 +12,7 @@ internal static class MessageContractHelper
     {
         internal static bool IsMessageContract(Type type)
         {
-            foreach (Attribute attr in type.GetCustomAttributes())
+            foreach (Attribute attr in type.GetCustomAttributes(inherit: false))
```

Comment:
Do you think there's any value before the call to return false to do the same call with `inherit: true` and if there's any matching attributes, throw an exception saying that MessageContractAttribute can't be applied to base types?

Reply:
I think that would be too aggressive behaviour here. That would make scenario of some type deriving from a chain of types which at some contains a MessageContractAttribute without having a MessageContractAttribute itself unsupported, and to me that would then imply that we expect that the type desires to use message contract if any of its base types have a MessageContractAttribute, which I am not sure is right, and would be a change of behaviour as compared to FX WCF.

Reply:
You convinced me, I agree.